backend/src/controllers/CMSTutorialController.ts

```diff
@@ -60,19 +61,22 @@ export default class TutorialController {
     /**
      * Updates a tutorial by it's {@link _Id ObjectId}
      *
+     * @async
      * @param {Request} req the users request obj
      * @param {Response} res our res obj
      */
     private static updateTutorialById(req: Request, res: Response): void {
-        if (TutorialController.validateTutorialData(req.body, 5)) {
+        if (CMSTutorialController.validateTutorialData(req.body, 5)) {
             const { _id, category, name, html, markdown } = req.body;
             const atomicDto = { $set: { ...new PartialTutorial(name, html, markdown, category) } };
             const predicateId = { _id: new ObjectId(_id) };
 
-            TutorialDbService.updateSingleDocument("tutorials", predicateId, atomicDto).then(
+            TutorialUpdateService.updateSingleDocument("tutorials", predicateId, atomicDto).then(
                 (resp) => res.status(200).json(resp).end(),
                 (err: MongoError) => res.status(503).send("Error name: " + err.name + "Code: " + err.code + "Msg: " + err.errmsg)
             );
+        } else {
+            res.status(400).send(BAD_REQUEST_TEXT).end();
         }
     }
```

Comment:
it's hard to figure out what that second parameter is on `validateTutorialData` see if you can be more declarative?
can you make use of template strings instead of doing it.
```js
res.status(503).send("Error name: " + err.name + "Code: " + err.code + "Msg: " + err.errmsg)
```

Reply:
Param is the key check, it's in the jsdoc for the method

And yeah I will, fair point lol. Realistically, it needs the object bits individually parsing into strings because I think the code can be object in some cases

Reply:
when I mean template strings I meant literals:
``` js
res.status(503).send(`Error name: ${err.name} Code: ${err.code} Msg: ${err.errmsg)

```
you can still you object this way.

Reply:
Yup, what I've responded with though is...

new GenerateError(code) being hooked into it, so it'll return a very specific stringified message in a solid format for the send().

Reply:
oh you have already done it and fixed it? is that what you mean?

Reply:
No no, I'll explain on call later aha. It's just an error generator to accomodate x/y/z format so our errors are consistent based on the err given by x/y/z service